spopt/tests/test_locate.py

```diff
@@ -375,26 +399,65 @@ def test_pmedian_client_facility_array_from_geodataframe(self):
             p_facilities=4,
         )
         pmedian = pmedian.solve(pulp.PULP_CBC_CMD(msg=False))
-        pmedian.facility_client_array()
-        pmedian.client_facility_array()
 
         numpy.testing.assert_array_equal(
             numpy.array(pmedian.cli2fac, dtype=object),
             numpy.array(pmedian_objective, dtype=object),
         )
 
+    def test_pmedian_preselected_facility_client_array_from_geodataframe(self):
+
+        known_objval = 1872.5093264630818
+        known_mean = 3.0299503664451164
+        known_solution_set = ["y_1_", "y_3_", "y_4_"]
+
+        fac_snapped = self.facilities_snapped.copy()
+        fac_snapped["predefined_loc"] = numpy.array([0, 0, 0, 1, 1])
+
+        pmedian = PMedian.from_geodataframe(
+            self.clients_snapped,
+            fac_snapped,
+            "geometry",
+            "geometry",
+            "weights",
+            predefined_facility_col="predefined_loc",
+            p_facilities=3,
+        )
+        pmedian = pmedian.solve(pulp.PULP_CBC_CMD(msg=False, warmStart=True))
+
+        observed_objval = pmedian.problem.objective.value()
+        assert known_objval == pytest.approx(observed_objval)
+
+        observed_mean = pmedian.mean_dist
+        assert known_mean == pytest.approx(observed_mean)
+
+        observed_solution_set = [dv.name for dv in pmedian.fac_vars if dv.varValue == 1]
+        numpy.testing.assert_array_equal(
+            numpy.array(known_solution_set, dtype=object),
+            numpy.array(observed_solution_set, dtype=object),
+        )
+
     def test_p_center_from_cost_matrix(self):
         p_center = PCenter.from_cost_matrix(self.cost_matrix, p_facilities=4)
         result = p_center.solve(pulp.PULP_CBC_CMD(msg=False))
         assert isinstance(result, PCenter)
 
+    def test_p_center_from_cost_matrix_no_results(self):
+        p_center = PCenter.from_cost_matrix(self.cost_matrix, p_facilities=4)
+        result = p_center.solve(pulp.PULP_CBC_CMD(msg=False), results=False)
+        assert isinstance(result, PCenter)
+
+        with pytest.raises(AttributeError):
+            result.cli2fac
+        with pytest.raises(AttributeError):
+            result.fac2clif
```

Comment:
Typo here. Instead `result.fac2clif` should be `result.fac2cli`. The same for lines 330, 196, 81.